app/android/src/uk/co/lutraconsulting/MMAndroidPosition.java

```diff
@@ -0,0 +1,190 @@
+import android.os.Looper;
+import android.os.Handler;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.Priority;
+
+
+
+public class MMAndroidPosition {
+
+    static public abstract class Callback {
+        public void onPositionChanged(@NonNull Location location, GnssStatus gnssStatus) {
+        }
+    }
+
+    private static native void jniOnPositionUpdated(int instanceId, Location location, GnssStatus gnssStatus);
+
+    static public MMAndroidPosition createWithJniCallback(Context context, boolean useFused, int instanceId) {
+        Log.i("CPP", "[java] createWithJniCallback");
+
+        MMAndroidPosition.Callback callback = new MMAndroidPosition.Callback() {
+            @Override
+            public void onPositionChanged(@NonNull Location location, GnssStatus gnssStatus) {
+                jniOnPositionUpdated(instanceId, location, gnssStatus);
+            }
+        };
+
+        return new MMAndroidPosition(context, callback, useFused);
+    }
+
+    private final Context mContext;
+    private final LocationManager mLocationManager;
+    private final boolean mUseFused;
+    private FusedLocationProviderClient mFusedLocationClient = null;
+    private final LocationCallback mLocationCallback;
+    private final LocationListener mLocationManagerCallback;
+    private final GnssStatus.Callback mGnssStatusCallback;
+    private final MMAndroidPosition.Callback mClientCallback;
+    private boolean mFusedAvailable = false;
+    private boolean mGpsProviderAvailable = false;
+    private boolean mIsStarted = false;
+    private String mErrorMessage;
+    private GnssStatus mLastGnssStatus;
+
+    public MMAndroidPosition(Context context, MMAndroidPosition.Callback clientCallback, boolean useFused) {
+        mContext = context;
+        mClientCallback = clientCallback;
+        mUseFused = useFused;
+
+        Log.i("CPP", "[java] constructor!");
+
+        mLocationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+
+        if (mUseFused) {
+            GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
+            mFusedAvailable = googleApiAvailability.isGooglePlayServicesAvailable(context) == ConnectionResult.SUCCESS;
+            Log.i("CPP", "[java] fused available: " + mFusedAvailable);
+            if (mFusedAvailable) {
+                mFusedLocationClient = LocationServices.getFusedLocationProviderClient(context);
+            }
```

Comment:
Shall we fallback to GPS if fused is not available?

Reply:
Updated the code to fall back to Qt implementation if fused is not available.